src/Reflection/ReflectionFunctionAbstract.php

```diff
@@ -442,11 +442,14 @@ public function getReturnType() : ?ReflectionType
             return null;
         }
 
+        $allowsNull = false;
+
         if ($returnType instanceof NullableType) {
             $returnType = $returnType->type;
+            $allowsNull = true;
```

Comment:
This should be an early return: overwriting variables is generally bad.

Reply:
Done.